### ISSUE TITLE
PSNotApplyErrorActionToStderr does not work as expected with ForEach-Object -Parallel

### ISSUE THREAD

**Issue by aelij** [PowerShell/PowerShell#14602]
## Steps to reproduce

~~Enable `PSNotApplyErrorActionToStderr`~~ - no longer needed

```powershell
$ErrorActionPreference = 'Stop'
@(1,2,3) | ForEach-Object -Parallel { cmd.exe /c "echo $_ >&2" }
```

## Expected behavior

Process all records, with no error records produced:

```none
1
2
3
```

## Actual behavior

Stops after first record, produces an error record for native command stderr:
```none
1 # in red
```

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.1.0
PSEdition                      Core
GitCommitId                    7.1.0
OS                             Microsoft Windows 10.0.20206
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by iSazonov** [comment]
It works well without `$ErrorActionPreference = 'Stop'`.

/cc @PaulHigin

**Comment by aelij** [comment]
> It works well without `$ErrorActionPreference = 'Stop'`.

Actually it doesn't. `Stop` only causes execution to continue, but it still creates error records (note the red output). If you remove `-Parallel`, no error records are created, as expected.

**Comment by PaulHigin** [comment]
This is a known issue.  Current session state is not propagated to parallel running sessions.  This is currently by design, but there is a work item to address this to some degree.

https://github.com/PowerShell/PowerShell/issues/12240

**Comment by aelij** [comment]
@PaulHigin How are experimental features flags related to a session? They are defined in `powershell.config.json`. Shouldn't every session load it independently?

**Comment by PaulHigin** [comment]
Sorry, I misread the issue.  I thought this was about propagating the $ErrorActionPreference built-in variable, and it looks like that is being done.  I am not sure about experimental features, but it is likely a similar problem in that it needs to be set in the parallel running sessions.

@aelij  Please add a comment to issue https://github.com/PowerShell/PowerShell/issues/12240, so that experimental feature enabling is also considered.  Then we can close this issue.

**Comment by aelij** [comment]
I think this should be a different issue.

This is about reading the correct config inside `Parallel`, not shared variables/session. It's likely the fix will be different too.

**Comment by PaulHigin** [comment]
It is the same in how the sessions are set up before running in parallel.  There is no point in having multiple issues.  I'll add a note to that relevant issue and close this one.

**Comment by aelij** [comment]
@PaulHigin I've investigated it a bit, and it doesn't seem that the issue is about experimental features at all:

```powershell
1 | ForEach-Object -Parallel { Get-ExperimentalFeature PSNotApplyErrorActionToStderr }
```

yields:
```
Name                                Enabled Source                              Description
----                                ------- ------                              -----------
PSNotApplyErrorActionToStderr          True PSEngine                            Don't have $ErrorActionPreference affect stderr output
```

I think the problem is in `NativeCommandProcessor` and how it handles output from background processes. So it should work regardless of the suggested `-UseCurrentState`.

This code gets executed only in parallel:

https://github.com/PowerShell/PowerShell/blob/4a09585fe72c3679e23a1aa047b23ac827609c55/src/System.Management.Automation/engine/NativeCommandProcessor.cs#L657-L673

So please reopen as it's definitely unrelated to the other issue.

**Comment by aelij** [comment]
Bump - still relevant

**Comment by RandyInMarin** [comment]
@aelij If the setting in the json file is changed, additional propagation from an active session should use the current session value. Using the new setting after a parallel pipeline has already started could lead to inconsistent behavior across threads.  If the new setting is required, start a new session.   As of 7.4, it's still manual labor to propagate preferences (e.g., WhatIf!) that don't propagate.  It's easier to propagate the current session preferences via using than to check if the default has changed.  So, for once, it's easier to do the right thing.

**Comment by aelij** [comment]
@RandyInMarin

> If the setting in the json file is changed, additional propagation from an active session should use the current session value.

Are you referring to the experimental flags JSON? That's not relevant anymore.

This issue is about how `-Parallel` handles `stderr` output differently. Just run the sample in the post.
